core/src/replay_stage.rs

```diff
@@ -2190,6 +2191,65 @@ impl ReplayStage {
                     .expect("All frozen banks must exist in the Progress map")
                     .computed;
                 if !is_computed {
+                    // Check if our tower is behind, if so (and the feature migration flag is in use)
+                    // overwrite with the newer bank.
```

Comment:
Maybe additional comment: We need to do this because the direct vote state instruction will fail if the updated state is missing a slot that has not yet expired in the on-chain state

Reply:
Chain looks like: 
1 -> 2 -> 3 -> 4 -> 5 -> 6 -> 7 -> 9

You voted 1 2 3 4 7
Local tower and on chain state: 1 2 3 7

Crashed, restart with partial local tower:
1 2 3

Our next vote is on 9, local tower looks like:

1 9

But this will be rejected by on chain state because 9 should not have popped off 2, 3 or 7 from the on chain state

Reply:
Hmmm now I'm thinking the check `if bank_vote_state.last_voted_slot()  > tower.vote_state.last_voted_slot()` might not be sufficient to prevent the above example if the votes from *before* the crash don't land until after we've restarted and voted for a *later* slot. 

For instance in the above example, assume none of your votes greater than `3` have landed *after* you restarted. Then your restarted local tower `1 2 3` passes the proposed check in this PR. Then you vote on `9` as in the above example. Then your previous votes land on chain.

Reply:
> 
> Crashed, restart with partial local tower: 1 2 3
> 
> Our next vote is on 9, local tower looks like: 1 9
> 

When we serialize/deserialize the tower the local vote state is also a part of it. This contains the lockouts no? So even if we restart with the partial tower it should be aware of 3's lockout and our local tower will look like 1 2 3 9

Reply:
can this inconsistency happen at any point other than restart? I think if we could sanitize the local tower during restart with something like 2. we could avoid this? instead of waiting for the tracer transaction could we do something with vote state on an optimistically confirmed block to avoid issues if roots are not being made?

Reply:
also in the worst case where we can't detect the inconsistency and we continue to vote, eventually we will build enough lockout that our votes will start landing right?